File: problems/58-length-of-last-word/csharp/58_length_of_last_word.cs
using System;

public static class Program
{
    public static void Main(string[] args)
    {
        Solution s = new Solution();
        Console.WriteLine(s.LengthOfLastWord("Hello World"));
        Console.WriteLine(s.LengthOfLastWord("   fly me   to   the moon  "));
        Console.WriteLine(s.LengthOfLastWord("luffy is still joyboy"));
    }
}

public class Solution
{
    public int LengthOfLastWord(string s)
    {
        s = s.Trim();
        string parsed = string.Empty;
        for (int i = s.Length - 1; i >= 0; i--)
        {
            if (char.IsWhiteSpace(s[i])) break;
            else parsed += s[i];
        }
        return parsed.Length;
    }
}

File: problems/58-length-of-last-word/python/58_length_of_last_word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        s = s.strip()
        parsed: str = ""
        for i in reversed(range(len(s))):
            if s[i] == " ":
                break
            parsed += s[i]
        return len(parsed)


def test(s: str) -> None:
    result: int = Solution().lengthOfLastWord(s)
    print(f"Length of the last word from '{s}' is {result}!")


if __name__ == "__main__":
    test("Hello World")
    test("   fly me   to   the moon  ")
    test("luffy is still joyboy")
